serialize.js

```diff
@@ -18,18 +22,21 @@ function serializeElement(elem) {
         tagname = tagname.toLowerCase()
     }
 
-    strings.push("<" + tagname +
-        properties(elem) + datasetify(elem) + ">")
+    strings.push("<" + tagname + properties(elem) + datasetify(elem))
 
-    if (elem.textContent || elem.innerText) {
-        strings.push(elem.textContent || elem.innerText)
-    }
+    if (voidElements.test(tagname)) {
+        strings.push(" />")
+    } else {
+        strings.push(">")
 
-    elem.childNodes.forEach(function (node) {
-        strings.push(serializeNode(node))
-    })
+        if (elem.childNodes.length) {
```

Comment:
Behavior isn't really defined if you have both childNodes and textContent/innerText set...previously it'd output both, which didn't really make sense, but now it prefers childNodes if there are any.

Probably never an issue in practice.